src/hooks/useSearchSubmit.ts

```diff
@@ -22,8 +23,22 @@ export function useSearchSubmit(): (query: string) => void {
       // Orders, transactions, tokens, batches
       const path = pathAccordingTo(query)
       const pathPrefix = prefixNetwork ? `${prefixNetwork}/${path}` : `${path}`
-
-      query && query.length > 0 && history.push(`/${pathPrefix}/${query}`)
+      if (pathPrefix === 'address' && query.match(/[a-zA-Z0-9]+\.[a-zA-Z]+$/)?.input !== undefined) {
+        if (web3) {
+          web3.eth.ens
+            .getAddress(query)
+            .then((res) => {
+              if (res && res.length > 0) {
+                history.push(`/address/${res}`)
+              } else {
+                history.push(`/address/${query}`)
+              }
+            })
+            .catch(() => history.push(`/address/${query}`))
```

Comment:
Keep the `${pathPrefix}` to keep the network in the query.

Unless you want to always ever redirect to mainnet address?

Reply:
Nope, just an error on my part in understanding how it works.  I added the `pathPrefix` back.